src/main/java/org/jitsi/videobridge/RtpChannel.java

```diff
@@ -2017,120 +1874,46 @@ public RtpChannelTransformEngine getTransformEngine()
         {
-            return;
+            return false;
         }
 
         this.setSources(sources); // TODO remove and rely on MSTs.
         this.setSourceGroups(sourceGroups); // TODO remove and rely on MSTs.
 
-        Map<Long, MediaStreamTrack> tracks = new TreeMap<>();
-        if (hasGroups)
-        {
-            List<SourceGroupPacketExtension> simGroups = new ArrayList<>();
-            Map<Long, Long> rtxPairs = new TreeMap<>();
-
-            for (SourceGroupPacketExtension sg : sourceGroups)
-            {
-                List<SourcePacketExtension> groupSources = sg.getSources();
-                if (groupSources == null || groupSources.isEmpty())
-                {
-                    continue;
-                }
-
-                if ("sim".equalsIgnoreCase(sg.getSemantics())
-                    && groupSources.size() >= 2)
-                {
-                    simGroups.add(sg);
-                }
-                else if ("fid".equalsIgnoreCase(sg.getSemantics())
-                    && groupSources.size() == 2)
-                {
-                    rtxPairs.put(
-                        groupSources.get(0).getSSRC(),
-                        groupSources.get(1).getSSRC());
-                }
-            }
+        MediaStreamTrackReceiver
+            mediaStreamTrackReceiver = stream.getMediaStreamTrackReceiver();
 
-            if (!simGroups.isEmpty())
-            {
-                for (SourceGroupPacketExtension simGroup : simGroups)
-                {
-                    MediaStreamTrack track = new MediaStreamTrack();
-
-                    int order = RTPEncoding.BASE_ORDER;
-                    for (SourcePacketExtension spe : simGroup.getSources())
-                    {
-                        Long primarySSRC = spe.getSSRC();
-                        Long rtxSSRC = rtxPairs.remove(primarySSRC);
-                        if (rtxSSRC != null)
-                        {
-                            track.addEncoding(primarySSRC, rtxSSRC, -1, order);
-                            tracks.put(primarySSRC, track);
-                            tracks.put(rtxSSRC, track);
-                        }
-                        else
-                        {
-                            track.addEncoding(primarySSRC, -1, -1, order);
-                            tracks.put(primarySSRC, track);
-                        }
-
-                        order++;
-                    }
-                }
-            }
-
-            if (!rtxPairs.isEmpty())
-            {
-                for (Map.Entry<Long, Long> fidEntry : rtxPairs.entrySet())
-                {
-                    MediaStreamTrack track = new MediaStreamTrack();
-                    Long primarySSRC = fidEntry.getKey();
-                    Long rtxSSRC = fidEntry.getValue();
+        if (mediaStreamTrackReceiver != null)
+        {
+            MediaStreamTrackImpl[] newTracks
+                = MediaStreamTrackFactory.createMediaStreamTracks(
```

Comment:
I was worried about what would happen when the objects are replaced, but it seems that setMediaStreamTracks takes care of that.